stored_requests/events/postgres/postgres.go

```diff
@@ -0,0 +1,78 @@
+package postgres
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/golang/glog"
+
+	"github.com/lib/pq"
+	"github.com/prebid/prebid-server/stored_requests/events"
+)
+
+type postgresEvents struct {
+	invalidations chan []string
+	updates       chan map[string]json.RawMessage
+}
+
+// NewPostgresEvents creates a new EventProducer listening to events on the given channel
+// via Postgres LISTEN/NOTIFY
+// Requires an event channel with the given name to exist on the database with payloads of the form:
+// json_build_object(
+//	'table',TG_TABLE_NAME,
+//	'action', TG_OP,
+//	'data', data
+// )
+// e.g.:
+// {
+//   "table": "stored_requests",
+// 	 "action": "INSERT",
+// 	 "data": {
+// 	   "id": "1",
+// 	   "requestData": "{\"id\": ...}"
+//   }
+// }
+func NewPostgresEvents(connInfo string, channel string, minReconnectInterval time.Duration, maxReconnectInterval time.Duration) (events.EventProducer, error) {
+	events := &postgresEvents{
+		invalidations: make(chan []string),
+		updates:       make(chan map[string]json.RawMessage),
+	}
+
+	reportProblem := func(ev pq.ListenerEventType, err error) {
+		if err != nil {
+			glog.Errorf("Error listening to Postgres notifications: %s", err.Error())
+		}
+	}
+
+	listener := pq.NewListener(connInfo, minReconnectInterval, maxReconnectInterval, reportProblem)
```

Comment:
Weird that you can only use a conn string and not just provide a db pointer to minimize number of connections. Could probably maintain a singleton `pq.Listener` shared between all `postgresEvents` instances and each maintains its separate `handleNotifications` goroutine as is, but that seems like an unnecessary early optimization.